sqlalchemy_redshift/dialect.py

```diff
@@ -216,10 +216,14 @@ def unquoted(self):
         even though the name must be quoted elsewhere.
         In particular, this happens for tables named as a keyword.
         """
-        key = str(self)
-        if key.startswith('"') and key.endswith('"'):
-            return key[1:-1]
-        return key
+        def unquote(part):
+            if (
+                    part is not None and part.startswith('"') and
+                    part.endswith('"')
+            ):
+                return part[1:-1]
+            return part
```

Comment:
This would feel more pythonic to me if we moved this to be a class method `_unquote` rather than defining this function nested inside the `unquoted` method. I'm not used to seeing nested functions.